lib/experimental/class-wp-rest-navigation-fallbacks-controller.php

```diff
@@ -0,0 +1,158 @@
+	 * Checks if a given request has access to read fallbacks.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
+	 */
+	public function get_item_permissions_check( $request ) {
+
+		$post_type = get_post_type_object( $this->post_type );
+
+		// Getting fallbacks requires creating and reading `wp_navigation` posts.
+		if ( ! current_user_can( $post_type->cap->create_posts ) || ! current_user_can( 'edit_theme_options' ) || ! current_user_can( 'edit_posts' ) ) {
+			return new WP_Error(
+				'rest_cannot_create',
+				__( 'Sorry, you are not allowed to create Navigation Menus as this user.', 'gutenberg' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		if ( 'edit' === $request['context'] && ! current_user_can( $post_type->cap->edit_posts ) ) {
+			return new WP_Error(
+				'rest_forbidden_context',
+				__( 'Sorry, you are not allowed to edit Navigation Menus as this user.', 'gutenberg' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Gets the most appropriate fallback Navigation Menu.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_item( $request ) {
+		$post = WP_Navigation_Fallbacks_Gutenberg::get_fallback_menu();
+
+		if ( empty( $post ) ) {
+			return rest_ensure_response( new WP_Error( 'no_fallback_menu', __( 'No fallback menu found.', 'gutenberg' ), array( 'status' => 404 ) ) );
+		}
+
+		$response = $this->prepare_item_for_response( $post, $request );
+
+		return $response;
+	}
+
+	/**
+	 * Retrieves the fallbacks' schema, conforming to JSON Schema.
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+		if ( $this->schema ) {
+			return $this->add_additional_fields_schema( $this->schema );
+		}
+
+		$this->schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'navigation-fallbacks',
+			'type'       => 'object',
+			'properties' => array(
+				'id' => array(
+					'description' => __( 'The unique identifier for the Navigation Menu.', 'gutenberg' ),
+					'type'        => 'integer',
+					'context'     => array( 'view', 'edit', 'embed' ),
+					'readonly'    => true,
+				),
+			),
+		);
+
+		return $this->add_additional_fields_schema( $this->schema );
+	}
+
+	/**
+	 * Matches the post data to the schema we want.
+	 *
+	 * @param WP_Post         $item The wp_navigation Post object whose response is being prepared.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response $response The response data.
+	 */
+	public function prepare_item_for_response( $item, $request ) {
+		$data = array();
+
+		$schema = $this->get_item_schema();
+
+		if ( isset( $schema['properties']['id'] ) ) {
```

Comment:
```suggestion
		$fields = $this->get_fields_for_response( $request );

		if ( rest_is_field_included( 'id', $fields ) ) {
```

Reply:
I'll update this 👍 

**Question**: is this pattern documented anywhere or does it require knowledge of the existing REST endpoints to be aware that it's a good practice? I asked because I [couldn't find any documentation covering it](https://developer.wordpress.org/rest-api/?s=rest_is_field_included).

Reply:
I've implemented this, but couldn't seem to get a test to trigger the `add_action` so I could `register_rest_field`.